pkg/platform/yaml/configdb/db.go

```diff
@@ -97,3 +98,62 @@ func (cdb *ConfigDb) setInstallationID(ctx context.Context, id string) error {
 
 	return nil
 }
+
+func (cdb *ConfigDb) ReadAccessToken(_ context.Context) (string, error) {
+	// read from cred.yaml
+	filePath := getTokenPath()
+
+	file, err := os.OpenFile(filePath, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		return "", fmt.Errorf("keploy access token file not found at %s", filePath)
+	}
+	defer func() {
+		err := file.Close()
+		if err != nil {
+			utils.LogError(cdb.logger, err, "failed to close file")
+		}
+	}()
+	decoder := yamlV.NewDecoder(file)
+	var apiKey string
+	err = decoder.Decode(&apiKey)
+	if errors.Is(err, io.EOF) || apiKey == "" {
+		return "", nil
+	}
+
+	return apiKey, nil
+}
+
+func getTokenPath() string {
+	path := UserHomeDir()
+	fileName := "token.yaml"
+	filePath := filepath.Join(path, fileName)
+	return filePath
+}
+
+func (cdb *ConfigDb) WriteAccessToken(ctx context.Context, token string) error {
+	filePath := getTokenPath()
+
+	// Check if the file exists; if not, create it
+	_, err := os.Stat(filePath)
+	if err != nil && os.IsNotExist(err) {
+		path := filepath.Dir(filePath)
+		_, err := yaml.CreateYamlFile(ctx, cdb.logger, path, "token")
+		if err != nil {
+			fileName := filepath.Base(filePath)
+			return fmt.Errorf("failed to create file %s. error: %s", fileName, err.Error())
+		}
+	} else if err != nil {
+		fileName := filepath.Base(filePath)
+		return fmt.Errorf("unable to access the keploy file %s. error: %s", fileName, err.Error())
+
+	}
+	d, err := yamlV.Marshal(&token)
+	if err != nil {
+		return fmt.Errorf("failed to marshal document to yaml. error: %s", err.Error())
+	}
+	err = os.WriteFile(filePath, d, os.ModePerm)
```

Comment:
I'm doubtful if this is even needed. We already create installation id, and this does not look more secure. 

How about we maintain a mapping of the installation ids to github ids on the backend? So for the same installation id, we could maintain the user and then only remove if someone wants to logout.

Reply:
If we do want to make it even more secure, we could as well just use machine unique id (instead of auto generated uuid stored in yaml) and maintain the mapping in the cloud service when someone tries to login. Maybe something like this -https://unix.stackexchange.com/questions/144812/generate-consistent-machine-unique-id